components/Card.js

```diff
@@ -16,31 +17,62 @@ export default function Card({
   return (
     <CardElement>
       {pathname === "/" && (
-        <DeleteButton aria-label="delete data" onClick={() => handleDelete(id)}>
+        <DeleteButton
+          aria-label="delete data"
+          onClick={() => handleDelete(id, "", "")}
+        >
           <Delete height={20} width={20} alt="delete" />
         </DeleteButton>
       )}
       <p>Date: {date}</p>
-      {weights.lenght !== 0 && (
+      {weights.length !== 0 && (
         <ul>
           {weights.map((weight) => (
-            <li key={weight._id}>Weight: {weight.value} </li>
+            <li key={weight._id}>
+              Weight: {weight.value}
+              {pathname === "/" && (
+                <DeleteSingleButton
+                  aria-label="delete single data point"
+                  onClick={() => handleDelete(id, weight._id, "weights")}
+                >
+                  <X height={20} width={20} alt="delete single data point" />
+                </DeleteSingleButton>
+              )}
+            </li>
           ))}
         </ul>
       )}
-      {heights.lenght !== 0 && (
+      {heights.length !== 0 && (
         <ul>
           {heights.map((height) => (
-            <li key={height._id}>Height: {height.value} </li>
+            <li key={height._id}>
+              Height: {height.value}
+              {pathname === "/" && (
+                <DeleteSingleButton
+                  aria-label="delete single data point"
+                  onClick={() => handleDelete(id, height._id, "heights")}
+                >
+                  <X height={20} width={20} alt="delete single data point" />
+                </DeleteSingleButton>
+              )}
+            </li>
           ))}
         </ul>
       )}
-      {feastTimes.lenght !== 0 && (
+      {feastTimes.length !== 0 && (
         <ul>
           {feastTimes.map((feastTime) => (
             <li key={feastTime._id}>
               Nurse time: {feastTime.value.substr(0, 2)}:
               {feastTime.value.substr(2, 2)}:{feastTime.value.substr(4, 2)}
+              {pathname === "/" && (
+                <DeleteSingleButton
+                  aria-label="delete single data point"
+                  onClick={() => handleDelete(id, feastTime._id, "feastTimes")}
+                >
+                  <X height={20} width={20} alt="delete single data point" />
+                </DeleteSingleButton>
+              )}
```

Comment:
Hi Mark, a small suggestion: you have used DeleteSingleButton three times, maybe it would be better when one uses it as a component so that the card page would have less code. The other looks good to me.